src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,217 @@
+            while (resultSet.next()) {
+                cars.add(parseCarWithManufacturerFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get all cars from DB", e);
+        }
+        for (Car car : cars) {
+            car.setDrivers(getDriversForCar(car.getId()));
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateRequest = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND is_delete = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateStatement
+                        = connection.prepareStatement(updateRequest)) {
+            updateStatement.setString(1, car.getModel());
+            updateStatement.setLong(2, car.getManufacturer().getId());
+            updateStatement.setLong(3, car.getId());
+            updateStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update a car " + car, e);
+        }
+        addDriverToCar(car);
+        removeDriverFromCar(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long carId) {
+        String deleteRequest = "UPDATE cars SET is_deleted = true WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement softDeleteCarStatement
+                        = connection.prepareStatement(deleteRequest)) {
+            softDeleteCarStatement.setLong(1, carId);
+            int numberOfDeleteRows = softDeleteCarStatement.executeUpdate();
+            return numberOfDeleteRows != 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete car with id " + carId, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getAllCarsRequest = "SELECT cars.id, cars.model, cars.manufacturer_id, "
+                + "manufacturers.name, manufacturers.country, "
+                + "FROM cars, cars_drivers, manufacturers "
+                + "WHERE cars_drivers.driver_id = ? AND cars.is_deleted = false;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllCarsStatement
+                        = connection.prepareStatement(getAllCarsRequest)) {
+            getAllCarsStatement.setLong(1, driverId);
+            ResultSet resultSet = getAllCarsStatement.executeQuery();
+            while (resultSet.next()) {
+                Car car = parseCarWithManufacturerFromResultSet(resultSet);
+                cars.add(car);
+            }
+            for (Car currentCur : cars) {
+                currentCur.setDrivers(getDriversForCar(currentCur.getId()));
+            }
```

Comment:
in this loop we have nested connection problem

Reply:
idk how we can do it in another way